### PR TITLE
Adds two tests on Transform+card changing effects

### PR DESCRIPTION
**Pull request by ssk97**
Testing edge cases for #11302 led to it failing the TransformTest with no easy solution to the problem, so I will be closing that PR. Perhaps some sort of batching of transforms would be possible, but the approach there seems nonviable currently.

That edge case also led me to realize that my prior work with SpellAbilityCastMode.getTypeModifiedCardObjectCopy also fell victim to a similar issue. While those changes were improvements in more common situations, it was a regression in this exact situation.

### CODE REVIEW

**Comment by ssk97** [comment]
To be more precise on the previous work:
Before those changes, a Baithook Angler//Hook-Haunt Drifter would be treated as a Baithook Angler until the cast was complete and the spell on the stack. After those changes it was properly treated as a Hook-Haunt Drifter during the cast, but card changing effects like Maskwood Nexus would not apply until the spell was on the stack.

File: Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/TransformTest.java [hunk 1]
```diff
@@ -494,4 +494,31 @@ public void testMoonmistDelver() {
...
+        addCard(Zone.BATTLEFIELD, playerA, "Maskwood Nexus"); //Make back side human
+
+
+        addCard(Zone.HAND, playerA, "Dress Down"); //Creatures lose all abilities
+        addCard(Zone.HAND, playerA, "Moonmist", 2);
+
+        castSpell(1, PhaseStep.UPKEEP, playerA, "Dress Down");
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Moonmist");
+
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/TransformTest.java:511]: could add a check in here to assure that it's back face between the two moonmists

**Review by xenohedron — APPROVED** [review]
